### ISSUE TITLE
`objc: Memory corruption in class list` crash on launch

### ISSUE THREAD

**Issue by MEHTAMAYUR** [firebase/firebase-ios-sdk#7651]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: _12.4____
  * Firebase SDK version: __Version 7.7.0 ___
  * Installation method: `CocoaPods ` 
  * Firebase Component: __GoogleUtilities/GULSwizzler___ (Auth, Core, Database, Firestore, Messaging, Storage, etc)

### [REQUIRED] Step 2: Describe the problem

On Launching of application,  Following method is called by APMScreenClassName load.

+(void) swizzleClass:(Class)aClass
            selector:(SEL)selector
     isClassSelector:(BOOL)isClassSelector
           withBlock:(nullable id)block

For selector. SEL	"viewDidDisappear:"  at the end of this method It crashes.

**With in main 0x000000016dd73730
objc[4836]: Memory corruption in class list.**

#### Steps to reproduce:

We are using 3rd party Library , after adding 3rd party Library in our code it is crashing on launch before calling Did finish launch method .

<img width="1455" alt="Screen Shot 2021-03-04 at 3 35 26 AM" src="https://user-images.githubusercontent.com/23650873/109887240-ccefaf80-7c9a-11eb-90ec-158db9c70114.png">
<img width="1523" alt="Screen Shot 2021-03-04 at 3 28 50 AM" src="https://user-images.githubusercontent.com/23650873/109887265-d7aa4480-7c9a-11eb-86a0-0e93e0e253dd.png">

**Comment by morganchen12** [comment]
This is almost definitely a result of Firebase classes being duplicated in your app's runtime. See the console warnings in your Xcode screenshot and check the full log output to see if there are multiple copies of Firebase in your app.

The error you're running into is defined in [`objc-runtime-new.mm`](https://opensource.apple.com/source/objc4/objc4-750/runtime/objc-runtime-new.mm.auto.html) and could occur if two copies of Firebase tried to swizzle the same class.

**Comment by morganchen12** [comment]
See this doc for more details: https://github.com/firebase/firebase-ios-sdk/blob/master/docs/firebase_in_libraries.md

**Comment by MEHTAMAYUR** [comment]
HI Team
Sorry for late reply.
I have removed firebase from pod framework. but now it is still giving  me same runtime error objc[18925]: Memory corruption in class list., but root cause is unkown.
This issue is only happens when I integrate 3rd party framework.
Previously I was able to find reason by keeping break point method swizzling  methods of GULSwizzler, but now There is no method swizzling in my project still same error.
